backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v4.ts

```diff
@@ -0,0 +1,938 @@
+import {
+  JsonController,
+  Post,
+  Body,
+  UseBefore,
+  Get,
+  BodyParam,
+  Req,
+  InternalServerError,
+  Delete,
+  Patch,
+  OnUndefined,
+} from 'routing-controllers';
+import { ExperimentService } from '../services/ExperimentService';
+import { ExperimentAssignmentService } from '../services/ExperimentAssignmentService';
+import { ExperimentAssignmentValidator } from './validators/ExperimentAssignmentValidator';
+import { ExperimentUser } from '../models/ExperimentUser';
+import { ExperimentUserService } from '../services/ExperimentUserService';
+import { UpdateWorkingGroupValidator } from './validators/UpdateWorkingGroupValidator';
+import {
+  IExperimentAssignmentv4,
+  ISingleMetric,
+  IGroupMetric,
+  SERVER_ERROR,
+  IGroupMembership,
+  IUserAliases,
+  IWorkingGroup,
+  PAYLOAD_TYPE,
+  EXPERIMENT_TYPE,
+} from 'upgrade_types';
+import { FailedParamsValidator } from './validators/FailedParamsValidator.v1';
+import { ExperimentError } from '../models/ExperimentError';
+import { FeatureFlag } from '../models/FeatureFlag';
+import { FeatureFlagService } from '../services/FeatureFlagService';
+import { ClientLibMiddleware } from '../middlewares/ClientLibMiddleware';
+import { LogValidator } from './validators/LogValidator';
+import { MetricService } from '../services/MetricService';
+import { ExperimentUserAliasesValidator } from './validators/ExperimentUserAliasesValidator';
+import { Metric } from '../models/Metric';
+import * as express from 'express';
+import { AppRequest } from '../../types';
+import { env } from '../../env';
+import { MonitoredDecisionPointLog } from '../models/MonitoredDecisionPointLog';
+import { MarkExperimentValidatorv4 } from './validators/MarkExperimentValidator.v4';
+
+interface ILog {
+  id: string;
+  uniquifier: string;
+  timeStamp: Date;
+  data: any;
```

Comment:
If this literally could be any unknown object, we should use `Record<string, unknown>` so that we still get typescript warnings about unknowable properties on this object potentially being undefined if we try to access it in code.